public/app/features/dashboard/components/PanelEditor/VisualizationSelectPane.tsx

```diff
@@ -103,6 +123,17 @@ export const VisualizationSelectPane = ({ panel, data }: Props) => {
                 onClose={() => {}}
               />
             )}
+            {listMode === VisualizationSelectPaneTab.Widgets && isWidgetEnabled && (
+              <VizTypePicker
+                current={plugin.meta}
+                onChange={onVizChange}
+                searchQuery={searchQuery}
+                data={data}
+                onClose={() => {}}
+                isWidget
+              />
+            )}
```

Comment:
I found a future issue here where, if the user has in the localStorage the widget listMode after enabling the feature flag, if the removes that feature flag he won't see any VizTypePicker.

```suggestion
            {listMode === VisualizationSelectPaneTab.Widgets && (
              <VizTypePicker
                current={plugin.meta}
                onChange={onVizChange}
                searchQuery={searchQuery}
                data={data}
                onClose={() => {}}
                isWidget={isWidgetEnabled}
              />
            )}
```

with this, we always show a list, despite the feature flag is disabled but the local storage has that listMode, yet

Reply:
Indeed, I added that extra condition intentionally. If the user does not have the feature flag this component should not be shown 😄.
🤔 I am failing to understand the future problem, maybe you can give me more details?

Reply:
sorry, my bad here. I thought the widget localStorage value was being retrieved although the user removed the feature flag. 
is the `isWidgetEnabled` necessary in `listMode === VisualizationSelectPaneTab.Widgets && isWidgetEnabled` ?
When you set the listMode in useState, you already ask for that boolean

Reply:
great catch 🙌🏾  , I think is not necessary, I will remove it!